File: app/src/main/java/com/example/administrator/shadowapplication/design_pattern/proxy/Shadow.java
package com.example.administrator.shadowapplication.design_pattern.proxy;

/**
 * Author : shadow
 * Desc :真实主题类，代理类所代表的真实主题
 * 客户端通过代理类间接的调用真实主题类的方法
 * Date :2018/4/12/012
 */

public class Shadow implements IShop {
    @Override
    public void buy() {
        System.out.println("购买红肠");
    }
}

File: app/src/main/java/com/example/administrator/shadowapplication/adapter/Shadow.kt
package com.example.administrator.shadowapplication.adapter

/**
 * @author 付影影
 * @desc
 * @date 2019/10/23
 */
class Shadow {
    var name: String? = null
        get() = if (field == null) "" else field
    var age: String = 0.toString()
}
